### ISSUE TITLE
Using library with DWM1000

### ISSUE THREAD

**Issue by ishirgarg** [thotro/arduino-dw1000#322]
I tried using this library with the DWM1000 module, however the Tag and Anchor programs did not seem to work upon upload: no communication occurred between the tag and anchor. I know for sure that the modules function properly as I am able to run my own test programs. Are there any modifications needed to this code so that it will work with the DWM1000, given that the tag/anchor programs are quite old?

**Comment by viaSeunghyun** [comment]
No, This Library is work.
I tested DWM1000 and Atmega328p.

Are you test DW1000Ranging_ANCHOR to DW1000Ranging_Tag or RangingAnchor to RangingTag?

**Comment by sharifsagar80** [comment]
I am also using trying to use the library but not working. I am using Arduino Uno and DWS1000 Arduino Shield. I tried both DW1000Ranging_ANCHOR to DW1000Ranging_Tag and RangingAnchor to RangingTag. I can get the device information in the serial monitor such as device mode and channel number like these.

> No, This Library is work.
> I tested DWM1000 and Atmega328p.
> 
> Are you test DW1000Ranging_ANCHOR to DW1000Ranging_Tag or RangingAnchor to RangingTag?

**Comment by Chrisrdouglas** [comment]
@sharifsagar80 I'm having a similar problem right now. did you ever get yours working? I'm able to get to the point where it prints the device address and ### ANCHOR ### or ### TAG ###

**Comment by sharifsagar80** [comment]
@Chrisrdouglas 
I changed the Arduino board with Nucleo f429zi. Now it is working well. you also need to redefine the reset and IRQ pin based on your microcontroller.

**Comment by Chrisrdouglas** [comment]
Oh dang, so you didn't get it to work with the arduino uno at all?

I've set the IRQ pin to be on pin 2 and the reset to be on pin 3

**Comment by ZakariaPZ** [comment]
@Chrisrdouglas Were you able to fix yours, and get it to work on the Uno?

**Comment by Chrisrdouglas** [comment]
No, I got super busy with work and haven't had the time to work on it in a while :/

Are you having similar problems? And did you try calibrating the antennas?

**Comment by ZakariaPZ** [comment]
@Chrisrdouglas I haven't actually started my project yet, but I've gotten some Unos and DWM1000s on order. I'm planning on following a certain tutorial setup, so I'm hoping it works out. Do you know how to calibrate the antennas?

**Comment by sharifsagar80** [comment]
@ZakariaPZ  Please refer to the DW1000 user manual v2. They give you a comprehensive description of the antenna delay calibration part.

**Comment by ZakariaPZ** [comment]
@sharifsagar80 Did you use this DW1000 library with the Nucleo?

**Comment by sharifsagar80** [comment]
Yes. I have used the DW1000 library with Nucleo.

**Comment by sharifsagar80** [comment]
> @sharifsagar80 Did you use this DW1000 library with the Nucleo?

Yes. I have used the DW1000 library with Nucleo
